core/amber/src/main/scala/edu/uci/ics/texera/web/resource/dashboard/project/ProjectResource.scala

```diff
@@ -69,6 +75,41 @@ object ProjectResource {
     )
   }
 
+  /**
+    * This method verifies a project exists with the corresponding
+    * pid, throwing an exception in the case it does not.
+    *
+    * @param pid project ID
+    */
+  private def verifyProjectExists(pid: UInteger): Unit = {
+    if (!userProjectDao.existsById(pid)) {
+      throw new BadRequestException("The project does not exist.")
+    }
+  }
+
+  /**
+    * This method verifies the user with the specified uid has access to
+    * the project with the specified pid, assuming such a project exists.
+    *
+    * If user has access, it will return the UserProject
+    * If not, it will throw a ForbiddenException stating insufficient access
+    *
+    * @param uid user ID
+    * @param pid project ID
+    * @return UserProject corresponding to pid
+    */
+  private def verifySessionUserHasProjectAccess(
+      uid: UInteger,
+      pid: UInteger
+  ): UserProject = {
+    val userProject: UserProject = userProjectDao.fetchOneByPid(pid)
+    if (userProject != null && userProject.getOwnerId != uid) {
+      // currently only owners should be able to access project
+      throw new ForbiddenException("No sufficient access privilege to project.")
+    }
+    userProject
+  }
+
```

Comment:
the method content is not exactly following the method name. it is doing 1) fetch a project 2) verify ownership 3) return the project. IMO if you call this method "verify..." it should just do 2).

Reply:
got it, updated function to only verify ownership, thanks!